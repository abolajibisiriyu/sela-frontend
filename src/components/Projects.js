import React from "react";

import ProjectsLayout from "../blocks/ProjectsLayout";
import Project from "./Project";

function Projects(props) {
  return (
    <ProjectsLayout.List>
      {(props.projects || []).map(project => (
        <Project key={project.id} project={project} />
      ))}
    </ProjectsLayout.List>
  );
}

export default Projects;
