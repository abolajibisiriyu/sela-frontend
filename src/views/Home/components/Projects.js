import React from "react";
import { connect } from "react-redux";

import ProjectsLayout from "../../../blocks/ProjectsLayout";
import SortBar from "../../../components/SortBar";
import Projects from "../../../components/Projects";
import Loader from "../../../components/Loader";

import { fetchProjects } from "../../../redux/projects/actions";
import H4 from "../../../elements/H4";

class HomeProjects extends React.Component {
  sortValues = [
    {
      value: "budget",
      text: "Budget"
    },
    {
      value: "startDate",
      text: "Start date"
    },
    {
      value: "endDate",
      text: "End Date"
    }
  ];
  constructor(props) {
    super(props);
    this.state = {
      sortValue: ""
    };
  }

  componentDidMount() {
    this.fetchProjects();
  }

  componentDidUpdate(prevProps, prevState) {
    if (prevState.sortValue !== this.state.sortValue) {
      this.fetchProjects();
    }
  }

  fetchProjects = () => {
    this.props.fetchProjects(this.state.sortValue);
  };

  onSortValuechanged = e => {
    this.setState({ sortValue: e.target.value });
  };

  render() {
    const { sortValue } = this.state;
    const { projects } = this.props;
    return (
      <div ref={this.props.projectsRef}>
        <ProjectsLayout>
          <ProjectsLayout.Header>
            <ProjectsLayout.HeaderSection>
              <h4>All Projects</h4>
            </ProjectsLayout.HeaderSection>
            <SortBar
              onChange={this.onSortValuechanged}
              sortValue={sortValue}
              sortValues={this.sortValues}
            />
          </ProjectsLayout.Header>
          {projects.fetching ? (
            <Loader />
          ) : projects.error ? (
            <H4>An Error Occured Fetching</H4>
          ) : (
            <Projects projects={projects.projects} />
          )}
        </ProjectsLayout>
      </div>
    );
  }
}

function mapStateToProps(state) {
  return {
    projects: state.projects
  };
}

export default connect(
  mapStateToProps,
  { fetchProjects }
)(HomeProjects);
