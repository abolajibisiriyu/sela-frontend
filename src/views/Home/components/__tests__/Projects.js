import React from "react";
import renderer from "react-test-renderer";
import configureMockStore from "redux-mock-store";
import { Provider } from "react-redux";
import thunk from "redux-thunk";
import promise from "redux-promise-middleware";

import Projects from "../Projects";

const middlewares = [thunk, promise];
const mockStore = configureMockStore(middlewares);
const store = mockStore({ projects: {} });

describe("Projects Component", () => {
  it("renders correctly", () => {
    const component = renderer.create(
      <Provider store={store}>
        <Projects />
      </Provider>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
