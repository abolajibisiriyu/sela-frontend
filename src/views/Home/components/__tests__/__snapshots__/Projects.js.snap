// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Projects Component renders correctly 1`] = `
<div>
  <section
    className="sc-bxivhb krOwCY"
  >
    <div
      className="sc-bdVaJa kIlXyG"
    >
      <div
        className="sc-bwzfXH eoSjYp"
      >
        <h4>
          All Projects
        </h4>
      </div>
      <div
        className="sc-gzVnrw pQFvb"
      >
        <div
          className="sc-ifAKCX gBnDME"
        >
          <h3>
            Sort by:
          </h3>
        </div>
        <div
          className="sc-bZQynM ikNgyi"
        >
          <select
            className="sc-EHOje jTMlhU"
            onChange={[Function]}
          >
            <option
              value=""
            >
              None
            </option>
            <option
              value="budget"
            >
              Budget
            </option>
            <option
              value="startDate"
            >
              Start date
            </option>
            <option
              value="endDate"
            >
              End Date
            </option>
          </select>
        </div>
      </div>
    </div>
    <div
      className="sc-htpNat jlmioB"
    />
  </section>
</div>
`;
